src/swapd/syncer_client.rs

```diff
@@ -283,6 +288,38 @@ impl SyncerState {
         self.tasks.tasks.insert(id, task.clone());
         task
     }
+    pub fn broadcast(&mut self, tx: bitcoin::Transaction) -> Task {
+        let id = self.tasks.new_taskid();
+        let task = Task::BroadcastTransaction(BroadcastTransaction {
+            id,
+            tx: bitcoin::consensus::serialize(&tx),
+        });
+        self.tasks.tasks.insert(id, task.clone());
+        self.tasks.broadcasting_txs.insert(id);
```

Comment:
I'm afraid there is redundancy here

tasks.tasks keeps track of ongoing tasks, and if the task is finished, the associated data should be removed

it looks like broadcasting_txs does exactly the same

Reply:
oh, i misread, its just the task id that is being stored, sorry for the confusion